### PR TITLE
Add shading of dependencies

### LINKED ISSUE #630: Shade the common dependencies in the library
Currently GitHub API declares several dependencies, including okhttp, commons-lang, commons-io, etc. These dependencies were updated in 1.99 by Dependabot, and it leads to risks when using the library in components which already include older versions of the libraries. 

One of the examples: Jenkins Core and GitHub API Plugin which bundles the library. https://github.com/jenkinsci/github-api-plugin/pull/24 updated the version there while Jenkins static checkers ignored it due to the exclusions in POM.xml: https://github.com/jenkinsci/github-api-plugin/blob/master/pom.xml#L77-L80 . Basically such approach may explode at any moment.

There are two possible solutions:

* Revert library updates in 1.99, pin the library versions
* Shade the dependencies so that GitHub API becomes more portable. I would recommend it though it may require a significant effort

### PR DESCRIPTION
**Pull request by bitwiseman**
Fixes #630

### CODE REVIEW

File: dependency-reduced-pom.xml [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <artifactSet>
+                <excludes>
+                  <exclude>com.infradna.tool:bridge-method-annotation</exclude>
+                  <exclude>org.jenkins-ci:annotation-indexer</exclude>
+                  <exclude>com.squareup.*:*</exclude>
```
Comment (bitwiseman) [inline dependency-reduced-pom.xml]: Optional dependency.

File: dependency-reduced-pom.xml [hunk 2]
```diff
@@ -0,0 +1,585 @@
...
+                  <exclude>org.jetbrains*:*</exclude>
+                  <exclude>com.github.spotbugs:*</exclude>
+                  <exclude>com.google.code.findbugs:*</exclude>
```
Comment (bitwiseman) [inline dependency-reduced-pom.xml]: compile time dependencies.

**Comment by bitwiseman** [comment]
Hm, thinking shading should only be done in ci and release maybe?

**Comment by bitwiseman** [comment]
@oleg-nenashev  Can you comment on this when you have time?

**Comment by oleg-nenashev** [comment]
I need to test it manually to comment. What is your plan for `dependency-reduced-pom.xml` ?

**Comment by bitwiseman** [comment]
@oleg-nenashev 
`dependency-reduced-pom.xml` get generated during build and is what is published on release.

**Comment by bitwiseman** [comment]
Ah, looks like it should be ignored. https://github.com/cloudera/director-google-plugin/issues/45

**Comment by bitwiseman** [comment]
@oleg-nenashev Updated.

**Pull request merged**